pkg/es/writer/esBulkHandler.go

```diff
@@ -318,3 +319,56 @@ func PostBulkErrorResponse(ctx *fasthttp.RequestCtx) {
 	responsebody["status"] = 400
 	utils.WriteJsonResponse(ctx, responsebody)
 }
+
+func ProcessDeleteIndex(ctx *fasthttp.RequestCtx, myid uint64) {
+	inIndexName := utils.ExtractParamAsString(ctx.UserValue("indexName"))
+
+	convertedIndexNames, indicesNotFound := deleteIndex(inIndexName, myid)
+
+	if indicesNotFound == len(convertedIndexNames) {
+		ctx.SetStatusCode(fasthttp.StatusNotFound)
+		response := make(map[string]interface{})
+		final := make(map[string]interface{})
+		var items = make([]interface{}, 0)
+		items = append(items, *utils.NewDeleteIndexErrorResponseInfo(inIndexName))
+		final["root_cause"] = items
+		final["error"] = *utils.NewDeleteIndexErrorResponseInfo(inIndexName)
+		final["status"] = 404
+		response["error"] = final
+		utils.WriteJsonResponse(ctx, response)
+	} else {
+		ctx.SetStatusCode(fasthttp.StatusOK)
+	}
+}
+
+func deleteIndex(inIndexName string, myid uint64) ([]string, int) {
+	convertedIndexNames := vtable.ExpandAndReturnIndexNames(inIndexName, myid, true)
+	indicesNotFound := 0
+	for _, indexName := range convertedIndexNames {
+
+		indexPresent := vtable.IsVirtualTablePresent(&indexName, myid)
+		if !indexPresent {
+			indicesNotFound++
+			continue
+		}
+
+		ok, _ := vtable.IsAlias(indexName, myid)
+		if ok {
+			alias, _ := vtable.GetAliasesAsArray(indexName, myid)
+			error := vtable.RemoveAliases(indexName, alias, myid)
```

Comment:
I think this is not right. If `indexName` is an alias, `ok==true` but then you call `GetAliasesAsArray(indexName, myid)`; shouldn't you call it on the non-aliased index name instead? Something like:
```
ok, baseIndexName := vtable.IsAlias(indexName, myid)
if ok {
	alias, _ := vtable.GetAliasesAsArray(baseIndexName, myid)
	error := vtable.RemoveAliases(baseIndexName, alias, myid)
...
```

Or do we want to delete just this alias but not the actual underlying index?